web/src/js/components/Widget/Widgets/Search/Search.js

```diff
@@ -38,12 +58,27 @@ class Search extends React.Component {
   }
 
   async executeSearch() {
-    const engine = this.state.config.engine || 'Google'
+    const engine = this.state.isYahooUser
+      ? 'Yahoo'
+      : this.state.config.engine || 'Google'
     const searchApi = this.getSearchApi(engine)
     const searchTerm = this.searchInput.input.value
-
-    // The page might be iframed, so opening in _top is critical.
-    window.open(searchApi + searchTerm, '_top')
+    //refetch latest local storage
+    const yahooHasAcknowledged = localStorageMgr.getItem(
+      STORAGE_YAHOO_SEARCH_DEMO
+    )
+    if (yahooHasAcknowledged === 'true') {
+      LogSearchMutation({
+        userId: this.props.user.id,
```

Comment:
This isn't working because the mutation, counterintuitively, takes the raw user ID rather than the Relay global. That's bad mutation design on my part as a workaround for another search landing page use case.

See [here](https://github.com/gladly-team/tab/blob/master/web/src/js/components/Search/SearchResults.js#L202) for an example of how to get it working. In our case, I think you'll want to fetch the user ID prior to the search being executed (probably on component mount) because the promise resolution might be canceled on page navigation (unsure).

Reply:
@kmjennison  locally verified that the mutation logs to table "userSearchLog".  I think that's right?  I am not currently awaiting the return of this mutation because I figure we want it to redirect asap, and regardless of whether the search log mutation returns back 200.  Is that ok?